File: src/com/designpatterns/creational/builder/Pizza.java
package com.designpatterns.creational.builder;

public class Pizza implements IPizzaMakingPlan {

    private Crust crustType;
    private int size;
    private boolean isNeedMozzarella;
    private boolean isNeedPepperoni;
    private boolean isNeedOlives;
    private boolean isNeedMushroom;
    private boolean isNeedCapsicum;
    private boolean isNeedOnions;

    @Override
    public void setCrustType(Crust crust) {
        crustType = crust;
    }

    @Override
    public void setSize(int noOfSlices) {
        size = noOfSlices;
    }

    @Override
    public void setMozzarella(boolean needMozzarella) {
        isNeedMozzarella = needMozzarella;
    }

    @Override
    public void setPepperoni(boolean needPepperoni) {
        isNeedPepperoni = needPepperoni;
    }

    @Override
    public void setOlives(boolean needOlives) {
        isNeedOlives = needOlives;
    }

    @Override
    public void setMushroom(boolean needMushroom) {
        isNeedMushroom = needMushroom;
    }

    @Override
    public void setCapsicum(boolean needCapsicum) {
        isNeedCapsicum = needCapsicum;
    }

    @Override
    public void setOnions(boolean needOnions) {
        isNeedOnions = needOnions;
    }

    @Override
    public String eatPizza() {
        return "crustType: "
                + crustType
                + "\n"
                + "size: "
                + size
                + "\n"
                + "isNeedMozzarella: "
                + isNeedMozzarella
                + "\n"
                + "isNeedPepperoni: "
                + isNeedPepperoni
                + "\n"
                + "isNeedOlives: "
                + isNeedOlives
                + "\n"
                + "isNeedMushroom: "
                + isNeedMushroom
                + "\n"
                + "isNeedCapsicum: "
                + isNeedCapsicum
                + "\n"
                + "isNeedOnions: "
                + isNeedOnions
                + "\n";
    }
}

File: src/com/designpatterns/creational/builder/kotlin/Pizza.kt
package com.designpatterns.creational.builder.kotlin

class Pizza : IPizzaMakingPlan {

    var crustType: Crust = Crust.FlatBreadCrust
    var size = 0
    var isNeedMozzarella = false
    var isNeedPepperoni = false
    var isNeedOlives = false
    var isNeedMushroom = false
    var isNeedCapsicum = false
    var isNeedOnions = false

    override fun eatPizza(): String {
        return """
            crustType: $crustType
            size: $size
            isNeedMozzarella: $isNeedMozzarella
            isNeedPepperoni: $isNeedPepperoni
            isNeedOlives: $isNeedOlives
            isNeedMushroom: $isNeedMushroom
            isNeedCapsicum: $isNeedCapsicum
            isNeedOnions: $isNeedOnions
            
            """.trimIndent()
    }
}